Readme.md

```diff
@@ -45,6 +45,15 @@ sent to the client (optional)
 * `contentType`: If the content to be sent is anything other than
 `application/json`, then the `contentType` property must be set (optional)
 * `bearerType`: string specifying the Bearer string (optional)
+* `function auth (key) {}` : this function will test if a key is valid. If the key is valid it
+must return a truthy value  or a promise that resolves to a truthy value. If function resolves to
+falsey values (false,null,undefined,0) authentication will fail with 401 value. If function fails
+or promise rejects authentication will fail (see `failSilent`). If `auth` is a function, `keys` will 
```

Comment:
Should we say that if auth is not a function it will be ignored?

Reply:
done